src/components/PopupWithForm.js

```diff
@@ -30,4 +34,23 @@ export default class PopupWithForm extends Popup {
     super.setEventListeners();
     this.formElement.addEventListener("submit", this._handleSubmitClick);
   }
+
+  setDeleteSubmitListener(data) {
+    this.data = data;
+    super.setEventListeners();
+    this.formElement.addEventListener("submit", this.deleteHandler);
+  }
+
+  deleteHandler() {
```

Comment:
There should **not** be any code in universal functions that doesn’t suit every popup.

`PopupWithForm` is for any modals with a form. It should not know about exact modals or actions